azurerm_role_definition: don't add `scope` to `assignable_scopes`

magodo opened the pull request:
fix #8577

magodo commented:
The two failures in the [test result](https://ci-oss.hashicorp.engineering/buildConfiguration/TerraformOpenSource_TerraformProviders_AzureRMPublic_AZURERM_SERVICE_PUBLIC_AUTHORIZATION/148472) seems unrelated to my changes.

jackofallops commented on website/docs/r/role_definition.html.markdown:
```diff
@@ -47,9 +47,7 @@ The following arguments are supported:
-* `assignable_scopes` - (Optional) One or more assignable scopes for this Role Definition, such as `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333`, `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup`, or `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM`.
-
-~> **NOTE:** The value for `scope` is automatically included in this list.
+* `assignable_scopes` - (Required) One or more assignable scopes for this Role Definition, such as `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333`, `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup`, or `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM`.
 
```
```suggestion
* `assignable_scopes` - (Optional) One or more assignable scopes for this Role Definition, such as `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333`, `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup`, or `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM`.
~> **NOTE:** The value for `scope` is automatically included in this list if no other values are supplied.
```

jackofallops reviewed (changes requested):
Hi @magodo 

Thanks for this PR, I've suggested a way to avoid a breaking change here that I believe should work correctly. It will require the tests to be updated also.

magodo commented:
@jackofallops Thank you for your comments, I have updated the code and added another test for the case that no assinable scopes is specified. Please take another look!

katbyte reviewed (approved):
Thanks @magodo - this LGTM 👍
